src/components/App.js

```diff
@@ -26,38 +26,35 @@ const SecondaryPanes = createFactory(require("./SecondaryPanes"));
 const WelcomeBox = createFactory(require("./WelcomeBox"));
 const EditorTabs = createFactory(require("./Editor/Tabs"));
 
-const App = React.createClass({
-  propTypes: {
-    sources: PropTypes.object,
-    selectSource: PropTypes.func,
-    selectedSource: PropTypes.object,
-    startPanelCollapsed: PropTypes.bool,
-    endPanelCollapsed: PropTypes.bool,
-  },
+class App extends React.Component {
+  state: {
+    horizontal: verticalLayoutBreakpoint.matches
+  }
 
-  displayName: "App",
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
@AnshulMalik I see, didn't know Flow was the reason for it. I think we could make the first one a type annotation for Flow like what is discussed [here](https://github.com/facebook/flow/issues/1594).